src/fides/api/service/connectors/s3_connector.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionTestStatus
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest, RequestTask
+from fides.api.schemas.connection_configuration.connection_secrets_s3 import S3Schema
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.query_config import QueryConfig
+from fides.api.util.aws_util import get_aws_session
+from fides.api.util.collection_util import Row
+from fides.connectors.models import ConnectorFailureException
+
+
+class S3Connector(BaseConnector):
+    """
+    AWS S3 Connector - this is currently used just to test connections to S3.
+
+    NOTE: No DSR processing is yet supported for S3.
+    """
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for s3"""
+        config = S3Schema(**self.configuration.secrets or {})
```

Comment:
if secrets only have an authentication method, nothing else, the Connection Test will be successful -

Reply:
OK! clearly i need to treat this function a bit better, i rushed it a bit - i'll dig into it further, thanks for highlighting.

Reply:
thanks for calling this out - my connection test was not a valid connection test! 😮 the command to retrieve buckets that i'd been calling just seemed to return an empty list with an invalid client, not actually fail.

i've adjusted the code to perform a better test - to get caller identity with sts. as the AWS documentatino notes, no permissions are required for this operation, and as far as i can tell, it's the generally recommended way to confirm valid credentials.

for what it's worth - if you have a valid `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` set in your `.env` (or just generally set as env vars in your container), then it's actually _expected_ that you'll be able to have a successful connection test using `automatic` authentication method with no credentials. i know the UX for that isn't the clearest, but i think it's acceptable for now. let me know if that doesn't make sense!

Reply:
ah okay thanks for explaining the `automatic` use case 👍